contracts/DKVDaggerHashimoto.sol

```diff
@@ -322,14 +371,18 @@ contract DecentralizedKVDaggerHashimoto is DecentralizedKV {
     }
 
     // We allow cross mine multiple shards by aggregating their difficulties.
+    // For some reasons, we never use checkIdList but if we remove it, we will get
+    // a `Stack too deap error`
     function mine(
         uint256 startShardId,
         uint256 shardLenBits,
         address miner,
         uint256 minedTs,
         uint256 nonce,
+        uint256[] memory checkIdList,
```

Comment:
It seems that the overflow error can be fixed by upgrading solidity to 0.8.16 (the latest hardhat supported version)

Reply:
Confirmed.